app/js/app/services/session.js

```diff
@@ -0,0 +1,342 @@
+    employmentStatuses: ['Full-time', 'Part-time', 'Self-Employed', 'Other Employment', 'None'],
+    educationLevels: ['Less than High School', 'High School / GED', 'Some College', '2-Year College Degree',
+      '4-Year College Degree', 'Master\'s Degree', 'PhD', 'Professional Degree'],
+    maritalStatuses: ['Single - Never Married', 'Married', 'Separated', 'Divorced', 'Widow', 'Life Partner', 'Dating Partner', 'Other'],
+    religions: ['Christian', 'Jewish', 'Muslim', 'Hindu', 'Buddhist', 'Atheist', 'Other'],
+    parties: ['Democrat', 'Republican', 'Independent', 'Libertarian', 'Socialist', 'Green, Other', 'Not American (Coming Soon)', 'I Don\'t Know'],
+    philosophies: ['Very Conservative', 'Conservative', 'Moderate', 'Liberal', 'Very Liberal', 'I Don\'t Know', 'Other'],
+    donors: ['Frequently', 'Occasionally', 'Never'],
+    registrations: ['Registered to Vote', 'Not Registered to Vote'],
+
+    states: ['AL', 'AK', 'AZ', 'AR', 'CA', 'CO', 'CT', 'DE', 'DC', 'FL', 'GA', 'HI', 'ID', 'IL', 'IN', 'IA', 'KS',
+      'KY', 'LA', 'ME', 'MD', 'MA', 'MI', 'MN', 'MS', 'MO', 'MT', 'NE', 'NV', 'NH', 'NJ', 'NM', 'NY', 'NC', 'ND',
+      'OH', 'OK', 'OR', 'PA', 'RI', 'SC', 'SD', 'TN', 'TX', 'UT', 'VT', 'VA', 'WA', 'WV', 'WI', 'WY', 'AS', 'GU',
+      'MP', 'PR', 'VI', 'UM'
+    ],
+
+    interests: ['Animals', 'Children', 'Community', 'Criminal Justice', 'Democracy', 'Economy', 'Education', 'Energy',
+      'Environment', 'Food', 'Healthcare', 'Human Rights', 'Infrastructure', 'LGBT', 'Labor', 'Leadership',
+      'NGOs', 'Peace', 'Politics', 'Poverty', 'Sustainability', 'Water', 'Women\'s Rights'],
+    countries: ['US']
+
+  };
+
+  return service;
+
+}).factory('facebook', function ($window, $q, $rootScope, iStorageMemory, stateAbbreviations,$facebook) {
+  var FB, uid, friends = [];
+
+
+  function loadProfile(params) {
+    var deferred = $q.defer();
+	
+	$facebook.api(uid + '/?fields=email,first_name,last_name,picture,gender,address,birthday,link').then( 
+	  function(fu) {
+		var data = {
+          facebook_id: params.facebook_id,
+          facebook_token: params.facebook_token,
+          country: 'US',
+          email: fu.email,
+          email_confirm: fu.email,
+          first_name: fu.first_name,
+          last_name: fu.last_name,
+          avatar_file_name: fu.pic,
+          sex: fu.gender ? fu.gender[0].toUpperCase() + fu.gender.slice(1, fu.gender.length) : '',
+          facebook_link: fu.link
+        };
+
+        if (fu.address) {
+          data.address1 = fu.address.street;
+          data.city = fu.address.city;
+          data.state = stateAbbreviations.US[fu.address.state];
+          data.zip = fu.address.zip;
+        }
+        if (fu.birthday) {
+          data.birth = fu.birthday;
+        }
+
+        if (fu.picture && fu.picture.data) {
+          data.avatar_file_name = fu.picture.data.url;
+        }
+
+        if (fu.email) {
+          data.username = fu.email.split('@')[0];
```

Comment:
@austinpapp What if jsmith@aol.com and jsmith@gmail.com are two different people? Will the system allow two usernames jsmith in the database or will it make sure there's a difference? The .split functionality is supposed to make it easier to create a Powerline username based on the Facebook user's e-mail address, but does the system accomodate for the jsmith@aol vs jsmith@gmail issue?